flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java

```diff
@@ -90,9 +92,9 @@ public void testFailedUpdateSubmission() throws Exception {
 
 	@Test
 	public void testSqlCompletion() throws IOException {
-		verifySqlCompletion("", 0, Arrays.asList("SELECT", "QUIT;", "RESET;"), Collections.emptyList());
-		verifySqlCompletion("SELEC", 5, Collections.singletonList("SELECT"), Collections.singletonList("QUIT;"));
-		verifySqlCompletion("SELE", 0, Collections.singletonList("SELECT"), Collections.singletonList("QUIT;"));
+		verifySqlCompletion("", 0, Arrays.asList("SOURCE", "QUIT;", "RESET;"), Collections.emptyList());
```

Comment:
Why drop the `SELEC` completion test ?

Reply:
before this pr, all commands are hint candidates. after this refactor,  only the commands who has regex pattern are hint candidates, or fallback to Table API hinting (will delegate to `tableEnv.getCompletionHints` in LocalExecutor). in `MockExecutor`, `completeStatement` method only returns `HintA` and `Hint B`